js/src/components/SchemaSettings.js

```diff
@@ -0,0 +1,163 @@
+import { Component, Fragment } from "@wordpress/element";
+import { __, _n, sprintf } from "@wordpress/i18n";
+import { Alert, HelpIcon, Select } from "@yoast/components";
+import PropTypes from "prop-types";
+import linkHiddenFields, { linkFieldsShape } from "./higherorder/linkHiddenField";
+import { schemaArticleTypeOptions, schemaPageTypeOptions } from "../values/SchemaOptions";
+
+/**
+ * Appends ' (default)' to the name of the default option.
+ *
+ * @param {Object[]} schemaTypeOptions The schema type options.
+ * @param {string} defaultType The default value to change the name for.
+ *
+ * @returns {Object[]} A copy of the schema type options.
+ */
+const addDefaultToOptionName = ( schemaTypeOptions, defaultType ) => {
+	const options = [];
+
+	// Clone the schema type options, but with the new name for the default.
+	schemaTypeOptions.forEach( ( option, index ) => {
+		options.push( {
+			value: option.value,
+			name: option.name,
+		} );
+
+		if ( option.value === defaultType ) {
+			options[ index ].name += ` (${ __( "default", "wordpress-seo" ) })`;
+		}
+	} );
```

Comment:
```suggestion
	schemaTypeOptions.forEach( ( option, index ) => {
		options.push( {
			value: option.value,
			name: option.name,
		} );

		if ( option.value === defaultType ) {
			options[ index ].name += ` (${ __( "default", "wordpress-seo" ) })`;
		}
	} );
```

is there a specific reason you are not using `.map`? Seems more readable to me.

Reply:
In your suggestion the option value is not cloned for non-default values, it should all be cloned to avoid strange bugs.